### DISCUSSION TITLE
Custom ID of new User

### DISCUSSION THREAD

**Discussion by stamartTCP** [keycloak/keycloak#20389] [Q&A] [answered]
I'm trying to create new user using API. But i want to assing custom UUID from my DB to keycloak user. But API ignore my setting and creates users with newly generated ID. Is it possible to do this? My Keycloak version is 21.1.1

here is my code:

`UserRepresentation user = new UserRepresentation();
    user.setId(userModel.getUUID().toString());
    user.setUsername(userModel.getEmail());
    user.setFirstName(userModel.getUserPersonalData().getFirstName());
    user.setLastName(userModel.getUserPersonalData().getLastName());
    user.setEmail(userModel.getEmail());
    user.setEnabled(true);
    
keycloak.realm(keycloakRealm).users().create(user)
    `

**Comment by sschu** [comment, marked as answer]
Currently, this is not possible.

**Reply by m1212e** [reply]
Is this planned or is there even a feature tracker I could subscribe to?

**Reply by sschu** [reply]
You can already import users providing their ID using the import feature. Simply providing the ID using the API will not be implemented as it allows reusing IDs of other user possibly taking over resources of these users outside of Keycloak, since user data is possibly linked to user ids there.

**Reply by m1212e** [reply]
I see thanks for the answer!
